tests/mail/tests.py

```diff
@@ -1605,12 +1612,220 @@ def test_8bit_non_latin(self):
         self.assertIn("Content-Transfer-Encoding: base64", txt.as_string())
 
 
+class WarnOnDeprecatedConfigurationTests(SimpleTestCase):
+    def test_deprecated_backend_configuration(self):
+        with self.assertWarnsMessage(
+            RemovedInDjango70Warning,
+            "EMAIL_BACKEND is deprecated. "
+            "Use EMAIL_PROVIDERS['default']['BACKEND'] instead.",
+        ):
+            with self.settings(
+                EMAIL_BACKEND="django.core.mail.backends.smtp.EmailBackend"
+            ):
+                pass
+
+    def test_deprecated_host_configuration(self):
+        with self.assertWarnsMessage(
+            RemovedInDjango70Warning,
+            "EMAIL_BACKEND is deprecated. "
+            "Use EMAIL_PROVIDERS['default']['BACKEND'] instead.",
+        ):
+            with self.settings(
+                EMAIL_BACKEND="django.core.mail.backends.smtp.EmailBackend",
+                EMAIL_HOST="smtp.example.com",
+            ):
+                connection = mail.get_connection()
+                self.assertIsInstance(connection, smtp.EmailBackend)
+                self.assertEqual(connection.host, "smtp.example.com")
+
+    @override_settings(
+        EMAIL_PROVIDERS={
+            "default": settings.EMAIL_PROVIDERS["default"],
+        },
+    )
+    def test_backend_configuration_mismatch(self):
+        with self.assertRaisesMessage(
+            ImproperlyConfigured,
+            "EMAIL_BACKEND and EMAIL_PROVIDERS are mutually exclusive.",
+        ):
+            with self.settings(
+                EMAIL_BACKEND="django.core.mail.backends.smtp.EmailBackend"
+            ):
+                pass
+
+    @override_settings(
+        EMAIL_PROVIDERS={
+            "default": settings.EMAIL_PROVIDERS["default"],
+        },
+    )
+    def test_host_configuration_mismatch(self):
+        with self.assertRaisesMessage(
+            ImproperlyConfigured,
+            "EMAIL_HOST and EMAIL_PROVIDERS are mutually exclusive.",
+        ):
+            with self.settings(
+                EMAIL_HOST="smtp.example.com",
+            ):
+                pass
+
+    def test_third_party_app_using_deprecated_default_configuration(self):
+        self.assertEqual(
+            settings.EMAIL_BACKEND,
+            "django.core.mail.backends.locmem.EmailBackend",
+        )
+        self.assertEqual(settings.EMAIL_HOST, "localhost")
+        self.assertEqual(settings.EMAIL_PORT, 25)
+
+    @override_settings(
+        EMAIL_PROVIDERS={
+            "default": {
+                "BACKEND": "django.core.mail.backends.dummy.EmailBackend",
+            },
+        },
+    )
+    def test_third_party_app_using_deprecated_backend_configuration(self):
+        self.assertEqual(
+            settings.EMAIL_BACKEND,
+            "django.core.mail.backends.dummy.EmailBackend",
+        )
```

Comment:
*Maybe* EMAIL_BACKEND is the exception to my earlier comment. But I can't think of any third party apps that would rely on settings.EMAIL_BACKEND being available like this. (It shouldn't be necessary for wrapper backends, and other apps would normally let get_connection() handle EMAIL_BACKEND.) 

Did you have some other example?

Reply:
This is just to test if all of the new configuration settings are passed as deprecated EMAIL_… settings to external 3rd party pass. And `EMAIL_BACKEND` is one of them.